### ISSUE TITLE
Adding another Subnet??? 

### ISSUE THREAD

**Issue by whoisibooo** [fizyr/keras-retinanet#1543]
Hi guys, 

i am looking for to add another subnet to the architecture . Do you have any ideas how I can archive that? 

best regards :)

**Comment by Ravitha** [comment]
Hi

To add a subnet, you have to modify the following scripts
1. retinanet.py  - Define the subnet and include the output of subnet as final model output
2. csv_generator.py - Load the subnet ground truth from excel file, process them and add it to the annotations dictionary
3. anchors.py - While calculating the target bounding box , ground truth for subnet output has to be computed
4. losses.py(optional) - add custom loss function for the subnet
5. train.py - Include the loss while compiling the model
6. In __init__ file of models package,add the new loss as custom object (optional)

Hope this is useful!

**Comment by whoisibooo** [comment]
Hi @Ravitha, 

thanks for your help. I will try to implement the following steps.

**Comment by abdussamettrkr** [comment]
Hi @whoisibooo ,
I added a new subnet to the Retinanet. If you wish, i could share that implementation with you.

**Comment by whoisibooo** [comment]
Hi @abdussamettrkr, 

for which task did you added a new subnet? yeah, it would be very kind of you. thanks

**Comment by abdussamettrkr** [comment]
> Hi @abdussamettrkr, 
> 
> for which task did you added a new subnet? yeah, it would be very kind of you. thanks 

My task have only 2 classes and data imbalance between classes is very high. So I added 2 new subnet (1 for classification and 1 for regression) each subnet group is predicting only one class objects. I will send you implementation in next hours.
